### ISSUE TITLE
Stuck in reconnect

### ISSUE THREAD

**Issue by GregorWaltzR** [FreeRDP/FreeRDP#9298]
In Windows 10 Pro, 3.0.0 wfreerdp.exe gets stuck in a reconnect loop:
[INFO][com.freerdp.client.common] - [client_auto_reconnect_ex]: Network disconnect!
[INFO][com.freerdp.client.common] - [client_auto_reconnect_ex]: Attempting reconnect (1 of 20)
numRetries is always 1. The client opens a black window.
I circumvented the issue by replacing the condition following freerdp_check_event_handles() with a continue:
```
client/Windows/wf_client.c 
1044,1049c1044
< 				if (client_auto_reconnect(instance))
< 					continue;
< 
< 				WLog_ERR(TAG, "Failed to check FreeRDP file descriptor");
< 				fflush(stdout);
< 				break;
---
> 				continue;
```
The error began on 8/5/2023 from master. The nightlies and my builds reflect this.
The SDL client also quit working on W10Pro with the same target some time after 8/4, but I do not see an obvious cause in the log.

**Comment by akallabeth** [comment]
@GregorWaltzR any more detailed information? There were a few windows updates that were buggy, so that might be the reason, but without some logs I can´t tell.

**Comment by GregorWaltzR** [comment]
Today's build works with the same conditions. I have attached a trace of the failing nightly from 8/16.
[wfreerdp.log](https://github.com/FreeRDP/FreeRDP/files/12413023/wfreerdp.log)

**Comment by akallabeth** [comment]
@GregorWaltzR ok, looks like the data is somewhere not being as expected (and, of course, one of the areas we´re currently not logging :/)

**Comment by akallabeth** [comment]
@GregorWaltzR I´ve created https://github.com/akallabeth/FreeRDP/tree/data_dump which dumps the packet data when the parsing fails.
would be nice if you could run a build from that and check if you have the data logged (I can then manually go through the parser and check where it fails)

**Comment by akallabeth** [comment]
@GregorWaltzR also created #9324 to log previously silent error conditions, so hopefully this will help to diagnose your issue.

**Comment by GregorWaltzR** [comment]
The data_dump build is able to connect just like the 8/22 nightly. I did not see the new logs from #9324 in the trace log (attached). As the current tree works, is the bug fixed or just in hiding? Would it be helpful to merge those log changes with my snapshot that manifests the bug?
[wfreerdp-data_dump.log](https://github.com/FreeRDP/FreeRDP/files/12432080/wfreerdp-data_dump.log)

**Comment by akallabeth** [comment]
@GregorWaltzR if you´re lucky yes. maybe a side effect fixed by https://github.com/FreeRDP/FreeRDP/pull/9308 ?

**Comment by GregorWaltzR** [comment]
> @GregorWaltzR if you´re lucky yes. maybe a side effect fixed by #9308 ?

That is it. I copied api.h and settings.c into the master from 8/15 and wfreerdp.exe now works.
Thanks
